src/pages/index.js

```diff
@@ -70,31 +127,113 @@ function createCard(data) {
   const cardElement = new Card(
     { data },
     selectors.cardTemplate,
-    handleImageClick
+    handleImageClick,
+    handleDeleteClick,
+    handleLikeClick,
+    handleDislikeClick
   );
   return cardElement.getView();
 }
 
 /*------- Event Handlers --------*/
 
-function handleImageClick(imgData) {
-  imageModal.open(imgData);
+function handleImageClick(title, image) {
+  imageModal.open(title, image);
 }
 
-function handleProfileFormSubmit(formValues) {
-  updateUserInfo(formValues);
+function handleProfileFormSubmit(inputs) {
+  const newName = inputs.name;
+  const newDescription = inputs.description;
+  profileEdit.renderSaving(true);
+  api
+    .updateUserInfo(newName, newDescription)
+    .then((response) => {
+      console.log("Profile updated successfully:", response);
+      currentUserInfo.setUserInfo({
+        name: newName,
+        description: newDescription,
+      });
+
+      profileEdit.close();
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    });
+  profileEdit.renderSaving(false);
+}
 
-  profileEdit.close();
+function handleAvatarFormSubmit(data) {
+  avatarModalPopup.renderSaving(true);
+  api
+    .updateAvatar(data)
+    .then((res) => {
+      console.log("Avatar updated successfully");
+      currentUserInfo.setAvatar(res.avatar);
+
+      avatarModalPopup.close();
+    })
+    .catch((error) => {
+      console.error("Error updating avatar:", error);
+    });
+  avatarModalPopup.renderSaving(false);
 }
 
-function handleAddCardFormSubmit(formValues) {
-  const { title: name, link } = formValues;
+function handleCardFormSubmit(inputValues) {
+  addCardModal.renderSaving(true);
+  api
+    .addCard({ name: inputValues.title, url: inputValues.url })
+    .then((cardData) => {
+      createCard(cardData);
+
+      section.renderCard(cardData);
+
+      addCardModal.close();
+    })
+    .catch((error) => {
+      console.error("Error adding card:", error);
+    });
+  addCardModal.renderSaving(false);
+}
+
+function handleDeleteClick(card) {
+  deleteModalPopup.open();
+  deleteModalPopup.setSubmitHandler(() => {
+    function deleteRequest() {
+      return api
+        .deleteCard(card._id)
+        .then(() => {
+          card.handleDelete();
+          console.log("Deleting card:", card);
+        })
+        .catch(console.error);
+    }
+    handleDeleteSubmit(deleteRequest);
+  });
+}
 
-  const cardElement = createCard({ name, link });
-  cardSection.addItem(cardElement);
+function handleDeleteSubmit(deleteRequest) {
+  deleteModalPopup.renderLoading(true);
+  deleteRequest()
+    .then(() => {
+      deleteModalPopup.close();
+    })
+    .catch(console.error);
 
-  addCardModal.resetForm();
-  addCardModal.close();
+  deleteModalPopup.renderLoading(false);
+}
+
+function handleLikeClick(card) {
+  api
+    .likeCard(card._id)
+    .then(card.toggleLikeBtn(), console.log(card._id + " has been liked"))
```

Comment:
`    .then(card.toggleLikeBtn(), console.log(card._id + " has been liked"))
`

In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 



```js


function handleLikeClick(card) {
  api
    .likeCard(card._id)
    .then(() => {
      card.toggleLikeBtn();
    })
    .catch(console.error);
}

```